doc/sphinx/proof-engine/proof-handling.rst

```diff
@@ -160,51 +174,34 @@ list of assertion commands is given in :ref:`Assertions`. The command
    the statement. In other words ``Proof using e`` is equivalent to
    ``Proof using Type + e`` for any declaration expression ``e``.
 
-   .. cmdv:: Proof using {+ @ident } with @tactic
-
-      Combines in a single line :cmd:`Proof with` and :cmd:`Proof using`.
-
-      .. seealso:: :ref:`tactics-implicit-automation`
+   .. todo: "All" should be in the grammar...
 
    .. cmdv:: Proof using All
 
-      Use all section variables.
+      :n:`All` (as an :n:`@ident`)
+        Use all section variables
 
-   .. cmdv:: Proof using {? Type }
+      :n:`Type {? * }`
+        Use only section variables occurring in the statement.  Specifying :n:`*`
+        uses the forward transitive closure of all the section variables occurring
+        in the statement. For example, if the variable ``H`` has type ``p < 5`` then
+        ``H`` is in ``p*`` since ``p`` occurs in the type of ``H``.
 
-      Use only section variables occurring in the statement.
+      :n:`{? * }`
+        Use the transitive closure of the specified collection.
 
-   .. cmdv:: Proof using Type*
+      :n:`- @ssexpr50`
+        Use all section variables except those specified by :n:`@ssexpr50`
 
-      The ``*`` operator computes the forward transitive closure. E.g. if the
-      variable ``H`` has type ``p < 5`` then ``H`` is in ``p*`` since ``p`` occurs in the type
-      of ``H``. ``Type*`` is the forward transitive closure of the entire set of
-      section variables occurring in the statement.
+      :n:`@ssexpr0 + @ssexpr0`
+        Use section variables from the union of both collections.
+        See :ref:`nameaset` to see how to form a named collection.
 
-   .. cmdv:: Proof using -({+ @ident })
-
-      Use all section variables except the list of :token:`ident`.
-
-   .. cmdv:: Proof using @collection__1 + @collection__2
-
-      Use section variables from the union of both collections.
-      See :ref:`nameaset` to know how to form a named collection.
-
-   .. cmdv:: Proof using @collection__1 - @collection__2
-
-      Use section variables which are in the first collection but not in the
-      second one.
-
-   .. cmdv:: Proof using @collection - ({+ @ident })
-
-      Use section variables which are in the first collection but not in the
-      list of :token:`ident`.
-
-   .. cmdv:: Proof using @collection *
-
-      Use section variables in the forward transitive closure of the collection.
-      The ``*`` operator binds stronger than ``+`` and ``-``.
+      :n:`@ssexpr0 - @ssexpr0`
+        Use section variables which are in the first collection but not in the
+        second one.
 
+      .. seealso:: :ref:`tactics-implicit-automation`
 
 Proof using options
```

Comment:
`@collection` should be changed to `@section_subset_expr` in `Default Proof Using` option below.